src/sentry/workflow_engine/models/detector_state.py

```diff
@@ -0,0 +1,27 @@
+from django.db import models
+
+from sentry.backup.scopes import RelocationScope
+from sentry.db.models import DefaultFieldsModel, FlexibleForeignKey, region_silo_model
+
+from .detector import Detector
+
+
+@region_silo_model
+class DetectorState(DefaultFieldsModel):
+    __relocation_scope__ = RelocationScope.Organization
+
+    class Type(models.TextChoices):
+        ACTIVE = "active"
+        INACTIVE = "inactive"
+
+    detector = FlexibleForeignKey(Detector)
+
+    # This key is used when a detector is using group-by
+    # allows us to link to a specific group from a single detector
+    detector_group_key = models.CharField(max_length=200, blank=True, null=True)
+
+    # If the status is currently active or not
+    status = models.CharField(max_length=200, choices=Type.choices)
+
+    # The current state of the detector
+    state = models.CharField(max_length=200, blank=True, null=True)
```

Comment:
Is the idea here that for a metric alert, we'd have
```
status: active, state: warning
status: active, state: critical
status: inactive, state: null?
```
or `status:ok/warning/critical`?

Reply:
The former, or close to it is what i was thinking:
```
status: active, state: warning
status: active, state: critical
status: inactive, state: ok
```

---

I was trying to design for a bunch of cases tho.. for performance issues, i was thinking:
```
status:inactive, state: ok
status:active, state: detected_n+1
```
Since those have an autoresolve, we could flip the `status` to `inactive` and leave the state:
```
status: inactive, state: detected_n+1
```
that could give us some other insights, and if we keep seeing the issue not be addressed we could add an escalation policy.

An escalation policy could be something like...
```
if status inactive -> active with !ok state
    update redis counter by detector / id.

if redis counter > escalation number
    trigger escalation
 ```

Reply:
^ pushed a quick update to it that i talked myself into in that comment 🤣 also addresses josh's concern about the status `Type` name.